File: ts-solutions/src/1431-kids-with-candies.ts
export function kidsWithCandies(
	candies: number[],
	extraCandies: number
): boolean[] {
	const m = Math.max(...candies)
	const result = new Array<boolean>(candies.length)
	for (let i = 0; i < candies.length; i++) {
		result[i] = candies[i] + extraCandies >= m
	}
	return result
}

File: go-solutions/1431_kids_with_candies.go
package leetcode

func KidsWithCandies(candies []int, extra int) []bool {
	maxCandies := -1
	for _, candy := range candies {
		maxCandies = max(maxCandies, candy)
	}

	result := make([]bool, len(candies))
	for i, candy := range candies {
		result[i] = candy+extra >= maxCandies
	}
	return result
}
